change_note_duration.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import fileinput
+import itertools
+import os
+import sys
+
+parser = argparse.ArgumentParser(description='Double or halve the note times in a file. '
+                                             'This does NOT change the time signature.',
+                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('file', help="Path to file where note values will be changed")
+parser.add_argument('-start_delimiter',
+                    default='Music',
+                    help='A string to search for in the file to indicate where note values will begin.')
+parser.add_argument('-end_delimiter',
+                    default='Verses',
+                    help='A string to search for in the file to indicate where note values end.')
+parser.add_argument('--double', action='store_true', help="Double note duration instead of halving it.")
+
+args = parser.parse_args()
+
+
+class NoteValueError(Exception):
+    pass
+
+
+def get_music_and_lyrics_line_numbers(file_name, start_delimiter, end_delimiter):
+    music_line = None
+    lyrics_line = None
+    with open(file_name, 'r') as file:
+        for (i, line) in enumerate(file):
+            if not music_line and start_delimiter in line:
+                music_line = i
+            if not lyrics_line and end_delimiter in line:
+                lyrics_line = i
+    return (music_line, lyrics_line,)
+
+
+def halve_note_duration(char):
+    try:
+       return str(int(char)*2)
+    except ValueError:
+        return char
+
+def double_note_duration(char):
+    try:
+        note_duration = int(char)
+        if note_duration == 1:
+            raise NoteValueError('Whole notes cannot be doubled')
+        else:
+            return str(note_duration // 2)
+    except ValueError:
+        return char
+
+
+def process_line(line, double_notes):
+    '''Double or halve note duration in a given line
+
+    Splits a string representing notes and note duration into the notes and rhythms,
+    and either halves or doubles those note durations.
+
+    :param line:
+    :param double_notes:
+    :return: new_line
+    '''
+    new_line = ''
+    for char in line:
+        if not double_notes:
+            new_line = "".join(["".join(x) if not is_digit else halve_note_duration("".join(x)) for is_digit, x in
+                                itertools.groupby(line, key=str.isdigit)])
+        else:
+            new_line = "".join(["".join(x) if not is_digit else double_note_duration("".join(x)) for is_digit, x in
+                                itertools.groupby(line, key=str.isdigit)])
```

Comment:
From a stylistic point of view, I would probably try to avoid duplicating the whole listcomp just to call one of two different functions. Instead I might pass the mutator function (`halve_note_duration` or `double_note_duration`) in as a parameter like so:

```
def process_line(line, mutator):
    return "".join(["".join(x) if not is_digit else mutator("".join(x)) for is_digit, x in
                                itertools.groupby(line, key=str.isdigit)])
```

(This is assuming I'm correct that the `for char in line` is left over and can be removed.)